### PR TITLE
add a new `@system-service` syscall group that is good as a starting point for whitelisting syscalls

### PR DESCRIPTION
**Pull request by poettering**
Previously we'd most blacklist some syscalls, but never whitelist, even though that's safer and most container environments do it.

This adds a new group syscall group `@system-service` that is supposed to be the starting point for the filters for typical system services. Note that `@system-service` is supposed to cover what typical, current system services need, and hence do not cover academic "ideal" cases, but real-world cases. i.e. setresuid() and friends are whitelisted, as daemons tend to call this even though in an ideal world `User=` would be a better approach.

This is then used to switch over all our long-running daemons to make use of this.

This stuff is to be seen in context with "portable services", i.e. #8620. as soon as both PRs are merged i'd like to turn on this new whitelist for the main portable service security profiles too.

The new filter group is inspired heavily by the policy nspawn applies by default, but removes everything needed only when spawning up full systems.

### CODE REVIEW

**Comment by poettering** [comment]
cc @tixxdz

**Comment by poettering** [comment]
Setting the "dont-merge" label for now, as I am trying to track down why the ubuntu CI fails with this

**Comment by poettering** [comment]
OK, dropping the "dont-merge" label. This passes CI very nicely on x86 and x86-64. s390 not so much, but given that it is currently flakey I am pretty sure it's unrelated to this patch.

PTAL!

File: man/systemd.exec.xml [hunk 1]
```diff
@@ -1462,6 +1462,10 @@ RestrictNamespaces=~cgroup net</programlisting>
                 <entry>@sync</entry>
                 <entry>Synchronizing files and memory to disk: (<citerefentry project='man-pages'><refentrytitle>fsync</refentrytitle><manvolnum>2</manvolnum></citerefentry>, <citerefentry project='man-pages'><refentrytitle>msync</refentrytitle><manvolnum>2</manvolnum></citerefentry>, and related calls)</entry>
               </row>
+              <row>
+                <entry>@system-service</entry>
+                <entry>A reasonable set of system calls used by common system services, excluding any special purpose calls. This is the recommended starting point for whitelisting system calls for system services, as it contains what is typically needed by system services, but excludes overly specific interfaces. For example, the following APIs are excluded: <literal>@clock</literal>, <literal>@mount</literal>, <literal>@swap</literal>, <literal>@reboot</literal>, …</entry>
```
Comment (keszybz) [inline man/systemd.exec.xml]: If "for example" then "…" is redundant.

File: src/portable/profile/strict/service.conf [hunk 2]
```diff
@@ -23,6 +23,8 @@ NoNewPrivileges=yes
 MemoryDenyWriteExecute=yes
 RestrictRealtime=yes
 RestrictNamespaces=yes
+SystemCallFilter=@system-service
+SystemCallErrorNumber=EPERM
```
Comment (keszybz) [inline src/portable/profile/strict/service.conf:27]: I don't understand the comment about "strict" profile in the commit message. It seems to be changed the same as others.

Comment (poettering) [reply src/portable/profile/strict/service.conf:27]: uh, that was supposed to say to exclude the "trusted" profile from this work. Will fix.

**Comment by poettering** [comment]
Force pushed a new version now. Not changes except the suggested man page fix and the correction of the commit msg. PTAL!

**Comment by keszybz** [comment]
LGTM.

**Pull request merged**